state-chain/runtime/src/chainflip.rs

```diff
@@ -819,11 +821,17 @@ impl OnBroadcastReady<Bitcoin> for BroadcastReadyProvider {
 				let output_len = outputs.len();
 				let vout = output_len - 1;
 				let change_output = outputs.get(vout).unwrap();
-				Environment::add_bitcoin_change_utxo(
-					change_output.amount,
-					UtxoId { tx_id, vout: vout as u32 },
-					batch_transfer.change_utxo_key,
-				);
+				if ScriptPubkey::Taproot(batch_transfer.change_utxo_key) ==
+					change_output.script_pubkey
+				{
+					Environment::add_bitcoin_change_utxo(
+						change_output.amount,
+						UtxoId { tx_id, vout: vout as u32 },
+						batch_transfer.change_utxo_key,
+					);
+				} else {
+					log_or_panic!("Keys do not match, no change_utxo available");
```

Comment:
That is a weird message and also shouldn't panic in debug? I think we should just log "Witnessed BTC egress without change UTXO"